pytorch/src/main/java/org/bytedeco/pytorch/presets/torch.java

```diff
@@ -2909,13 +100,50 @@ public class torch implements LoadEnabled, InfoMapper {
         Loader.checkVersion("org.bytedeco", "pytorch");
     }
 
+    static void initIncludes(Class thisClass, ClassProperties properties) {
+        // If we are called from Parser, fetch the list of headers to parse from resources
+        if (!Loader.isLoadLibraries()) {
+            for (int i = 3; i < 10; i++) {
+                Class caller = Loader.getCallerClass(i);
```

Comment:
You don't need to have a loop here. Just use 3 if that works, and if it changes later, which it probably wont, but if it does we can always easily fix that.

Reply:
I do because the index of Parser in the stack trace is not the same when coming from torch parsing or torch_cuda parsing.

Reply:
Right, but you're also saying you don't want the same list anyway for torch and torch_cuda, so why do you want to  catch both here when in the line right after that you only proceed if it's not torch_cuda?

Reply:
When we parse, init is called at lease 3 times:
1. torch.init() when processing torch presets
2. torch_cuda.init() when processing torch_cuda presets, 
3. torch.init() when processing torch_cuda presets (because of properties inheritance)

We need at least to empty the include property in 3 cases, so we need to check if we are called from Parser in 3 cases. And parser position is 5 for 1. and 2. and 6 for 3.. Then we want to fill the property with the resource file in cases 1. and 2. only

I don't think we can do much simpler.
The `isLoadLilbraries()` test at the start of `initIncludes` is useless though.

Reply:
However this code loads uselessly the big Parser  class when the users load the presets. Maybe we'd better test for the class name instead of `== Parser.class`, or add something to parser to avoid querying the stack track, like using a distinct initParsing() method.

Reply:
> I don't think we can do much simpler.

I'm pretty sure Loader.getCallerClass(3) is going to return the Parser class when we want it, in both torch.init() and torch_cuda.init(), and return something else when we don't want it anyway, so I don't think we need a loop. Just please try it!

Reply:
Here are the stack traces.
```
CALLED FROM torch with effective class=torch
java.lang.Exception
        at org.bytedeco.pytorch.presets.torch.initIncludes(torch.java:107)
        at org.bytedeco.pytorch.presets.torch.init(torch.java:144)
        at org.bytedeco.javacpp.ClassProperties.load(ClassProperties.java:326)
        at org.bytedeco.javacpp.Loader.loadProperties(Loader.java:386)
        at org.bytedeco.javacpp.tools.Parser.parse(Parser.java:4343)
```
```
CALLED FROM torch with effective class=torch_cuda
java.lang.Exception
        at org.bytedeco.pytorch.presets.torch.initIncludes(torch.java:107)
        at org.bytedeco.pytorch.presets.torch.init(torch.java:144)
        at org.bytedeco.javacpp.ClassProperties.load(ClassProperties.java:326)
        at org.bytedeco.javacpp.ClassProperties.load(ClassProperties.java:162)
        at org.bytedeco.javacpp.Loader.loadProperties(Loader.java:386)
        at org.bytedeco.javacpp.tools.Parser.parse(Parser.java:4343)
```
```
CALLED FROM torch_cuda with effective class=torch_cuda
java.lang.Exception
        at org.bytedeco.pytorch.presets.torch.initIncludes(torch.java:107)
        at org.bytedeco.pytorch.presets.torch_cuda.init(torch_cuda.java:68)
        at org.bytedeco.javacpp.ClassProperties.load(ClassProperties.java:326)
        at org.bytedeco.javacpp.Loader.loadProperties(Loader.java:386)
        at org.bytedeco.javacpp.tools.Parser.parse(Parser.java:4344)
```
We can write 
```Java
if (Loader.getCallerClass(5).getName().equals("org.bytedeco.javacpp.tools.Parser") || Loader.getCallerClass(6).getName().equals("org.bytedeco.javacpp.tools.Parser")) 
```
instead if you prefer. 
But what bothers you exactly with this loop ? It seems more resilient in the long run to small modifications of javacpp.

Reply:
Ah, I think I understand what you mean : you would like to use `Loader.getCallerClass(5).getName().equals("org.bytedeco.javacpp.tools.Parser")` to test both that we are called from Parser and that we are not in the case `properties.getEffectiveClasses().get(0) != thisClass`.
The problem (besides code resiliance) is that, as said above, we must reset the include property (`properties.put("platform.include", new ArrayList<String>())`) when we are called by the parser in all cases.

Reply:
> 

> The problem (besides code resilience) is that, as said above, we must reset the include property (`properties.put("platform.include", new ArrayList<String>())`) when we are called by the parser in all cases.

No we don't. What happens if you don't do that?

Reply:
Ok, I see: resetting the properties only once when processing torch_cuda is enough.
So we can replace the loop and the check for effective class by a check for depth 5 only. It's a bit cryptic but there will be less lines of code.